RecoParticleFlow/PFProducer/interface/KDTreeLinkerBase.h

```diff
@@ -23,6 +23,10 @@ class KDTreeLinkerBase {
 
   void setFieldType(const reco::PFBlockElement::Type &fld) { _fieldType = fld; }
 
+  // set trajectory points, necessary for track-HCAL links
+  virtual void setTrajectoryPoints(const reco::PFTrajectoryPoint::LayerType trajectoryPointEntrance,
+                                   const reco::PFTrajectoryPoint::LayerType trajectoryPointExit){};
```

Comment:
looking at the use cases, I do not see a good justification to add this method at the base class level.
It seems more practical to me to use a more common pattern of passing configuration to the tool via a PSet.
I suggest to add a ```KDTreeLinkerBase(const edm::ParameterSet& conf)``` and add the appropriate logic to KDTreeLinkerTrackHcal(conf) constructor.

See more in comments for PFBlockAlgo.cc